data-pipeline/csv-validation/petl-validation.py

```diff
@@ -0,0 +1,51 @@
+
+# Exception Classes
+class InvalidStateException(Exception):
+  pass
+class InvalidJurisdictionException(Exception):
+  pass
+
+class InvalidCountyException(Exception):
+  pass
+
+# Validators 
+def validate_state(val):
+  if val != "HI":
+    raise InvalidStateException
+
+# TODO: Verify jurisdiction based on the file passed in
+# TODO: Account for okinas in Hawaii? 
+def validate_jurisdiction(val):
+  jurisdictions = ["Hawaii", "Kauaʻi", "Maui", "Honolulu"]
+  if val not in jurisdictions:
+    raise InvalidJurisdictionException
+
+# TODO: Verify jurisdiction based on the file passed in
+# TODO: Account for okinas in Hawaii? 
+def validate_county(val):
+  counties = ["Hawaii", "Kauaʻi", "Maui", "Honolulu"]
+  if val not in counties:
+    raise InvalidCountyException
+
+# TODO: Determine constraints for abbreviated district names
+# TODO: Verify that abbreviated district names can't be only numbers
+def validate_abbr_district_name(val):
+  if not isinstance(val,str) or val.isnumeric():
+    raise ValueError
+
+constraints = [
+    dict(name='state_test', field='State', test=validate_state),
+    dict(name='jurisdiction_test', field='Jurisdiction', test=validate_jurisdiction),
+    dict(name='county_test', field='County', test=validate_county),
+    dict(name='abbr_district_name_test', field='Abbreviated District Name', test=validate_abbr_district_name),
+]
+
+table_full = etl.fromcsv('../hawaii-zoning-data.csv')
+table = etl.tail(table_full, table_full.len() - 2)
+problems = etl.validate(table, constraints=constraints, header=headers)
+if problems.len() > 2:
```

Comment:
Why the `> 2` check here?

Reply:
The > 2 is because there is an  AssertionError on the header line for some reason.  The length without any errors was 2 when I checked, but fixing whatever causes that AssertionError would most likely fix the 2 magic number.